### DISCUSSION TITLE
How long should a 1 hour 38 minute podcast take to transcribe?

### DISCUSSION THREAD

**Discussion by mjyoung** [deepgram/community#551] [General help] [answered]
Hello, new user here.

I'm using `nova-2-meeting` and attempting to transcribe a podcast to test this service out. I have the All-In podcast encoded to a 12k bitrate ogg file [here](https://storage-dev.romeano.com/uploads/misc/ALLIN-E162_12k.ogg). This 1 hour 38 minute podcast has a filesize of 9mb. 

It has been `Pending` in the logs for almost 30 minutes now. 

I tested this with a smaller/shorter recording (30s) and it processed fine and hit the callback URL with the results as expected.

My request id if it helps: `27dadf9d-9fcd-4bd1-89b4-efed3fc21a95`

This is my JS SDK code:
```
const transcribeAudioDeepgram = async (audioUrl: string) => {
  try {
    const { result, error } =
      await deepgram.listen.prerecorded.transcribeUrlCallback(
        {
          url: audioUrl,
        },
        new CallbackUrl(<webhook_url>),
        {
          model: "nova-2-meeting",
          language: "en",
          alternatives: 1,
          diarize: true,
          detect_entities: true,
          detect_topics: true,
          paragraphs: true,
          profanity_filter: false,
          smart_format: true,
          summarize: true,
        },
      );
    if (error) {
      console.error(error);
      return error;
    }
    return result;
  } catch (err) {
    console.error(err);
    return err;
  }
};
```

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by mjyoung** [comment]
Over an hour and a half later and still not done. Guess it's stuck in `Pending` purgatory

**Comment by mjyoung** [comment]
Hmmm okay commented out a bunch of features and it completed pretty quickly:

```js
  const { result, error } =
      await deepgram.listen.prerecorded.transcribeUrlCallback(
        {
          url: audioUrl,
        },
        new CallbackUrl(
          "<webhook_url>",
        ),
        {
          model: "nova-2-meeting",
          language: "en",
          // alternatives: 1,
          diarize: true,
          // detect_entities: true,
          // detect_topics: true,
          // paragraphs: true,
          // profanity_filter: false,
          smart_format: true,
          // summarize: true, // summarize costs more
        },
      );
```

**Comment by jkroll-deepgram** [comment]
Hi @mjyoung, I'm sorry that your request wasn't completing in a timely manner. Thanks for sharing that original request ID of `27dadf9d-9fcd-4bd1-89b4-efed3fc21a95`. I do see that we received the request but did not return a response. On rare occasions, a request may get "stuck" as you said, where there is a silent failure, and a reflection of "Pending" in the console. After more time elapses, it will reflect as "Lost," and you will not be charged for it.

Did you see this behavior on any subsequent API calls? Your parameters all look fine, and shouldn't cause any particular problems.

One other small observation, do note that the profanity filter is only available on our `base` model, not any of the `nova-2` models. The other parameters will work with the `nova-2-meeting` model you're using.

**Reply by mjyoung** [reply]
Hi @jkroll-deepgram, I have another request ID:
`966a12f1-3f67-4994-9d87-c23d5a7147ce`

With these params:
`POST /v1/listen?model=nova-2-meeting&language=en&smart_format=true&diarize=true&alternatives=1&detect_entities=true&detect_topics=true&paragraphs=true&summarize=true&callback=<webhook_url>`

Started this one ten minutes ago with the same file as previous and it appears to be timing out as well. 

Again the file I am testing with is [this one](https://storage-dev.romeano.com/uploads/misc/ALLIN-E162_12k.ogg)

**Reply by mjyoung** [reply]
Just tried another request, this time without `detect_entities`:
Request ID `fdb751a2-2e27-4515-8a78-8eafca661908`

`POST /v1/listen?model=nova-2-meeting&language=en&smart_format=true&diarize=true&alternatives=1&detect_topics=true&paragraphs=true&summarize=true&callback=<webhook_url>`

This one completed after 2 minutes. Are you aware of any issues with `detect_entities` causing long transcripts to time out?

**Reply by jkroll-deepgram** [reply, marked as answer]
Hi @mjyoung, apologies for the belated reply. I see you're using a few different of our [intelligence](https://developers.deepgram.com/reference/audio-intelligence-apis) features (detecting entities and topics, and summarization). Each of these features is more compute-intensive than simple ASR, so that will add additional layers of latency and potential points of failure.

I'm curious how this has been going for you in the past week? Have you found it feasible to use all three of these parameters in most requests?

The intelligence features are newer, and we're still establishing latency norms in production. I'd appreciate being able to learn from your experience and share your feedback internally. Are you finding it usable, or that you need to pare down your features or run multiple API calls on the same audio for getting all of the intelligence feature outputs?
